pkgs/development/python-modules/pandas/default.nix

```diff
@@ -55,51 +62,47 @@ buildPythonPackage rec {
     xlwt
   ];
 
-  checkInputs = [ pytestCheckHook glibcLocales hypothesis ];
+  checkInputs = [
+    glibcLocales
+    hypothesis
+    jinja2
+    pytest-asyncio
+    pytest-xdist
+    pytestCheckHook
+    XlsxWriter
+  ];
 
-  # doesn't work with -Werror,-Wunused-command-line-argument
+  # Doesn't work with -Werror,-Wunused-command-line-argument
   # https://github.com/NixOS/nixpkgs/issues/39687
   hardeningDisable = lib.optional stdenv.cc.isClang "strictoverflow";
 
-  # Parallel Cythonization is broken in Python 3.8 on Darwin. Fixed in the next
-  # release. https://github.com/pandas-dev/pandas/pull/30862
-  setupPyBuildFlags = lib.optionals (!(isPy38 && stdenv.isDarwin)) [
-    # As suggested by
-    # https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#creating-a-python-environment
-    "--parallel=$NIX_BUILD_CORES"
-  ];
+  # For OSX, we need to add a dependency on libcxx, which provides
+  # `complex.h` and other libraries that pandas depends on to build.
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    cpp_sdk="${libcxx}/include/c++/v1";
+    echo "Adding $cpp_sdk to the setup.py common_include variable"
+    substituteInPlace setup.py \
+      --replace "['pandas/src/klib', 'pandas/src']" \
+                "['pandas/src/klib', 'pandas/src', '$cpp_sdk']"
+  '';
 
   doCheck = !stdenv.isAarch64; # upstream doesn't test this architecture
 
   pytestFlagsArray = [
     "--skip-slow"
     "--skip-network"
```

Comment:
```suggestion
    "--skip-network"
    "-n auto" # or any other sensible value that works
```

Reply:
Should not that rather be `"-n" "auto"`?

Reply:
Several times I've had failing tests with pandas due to xdist so I suggest setting `-n0`

Reply:
> Should not that rather be `"-n" "auto"`?

What is the difference?

Reply:
> Several times I've had failing tests with pandas due to xdist so I suggest setting -n0, thereby disabling concurrent tests.

Then we can remove xdist entirely.

> What is the difference?

Quoting by bash and how the program splits its args.

Reply:
Following @FRidh's input here.

Reply:
> What is the difference?

-n0 disables parallel execution, auto adds a process for each core.

> Then we can remove xdist entirely.

Unfortunately not. A fixture that is provided by pytest-xdist is used.